arangod/Pregel/GraphStore/GraphStore.cpp

```diff
@@ -766,48 +518,32 @@ void GraphStore<V, E>::storeResults(
   double now = TRI_microtime();
   TRI_ASSERT(SchedulerFeature::SCHEDULER != nullptr);
 
-  size_t const numSegments = _vertices.size();
-
-  uint32_t numThreads = 1;
-  if (_localVertexCount > 100000) {
-    // We expect at least parallelism to fit in a uint32_t.
-    numThreads = static_cast<uint32_t>(
-        std::min<size_t>(_config->parallelism(), numSegments));
-  }
-
-  _runningThreads.store(numThreads, std::memory_order_relaxed);
-  _feature.metrics()->pregelNumberOfThreads->fetch_add(numThreads);
-  size_t const numT = numThreads;
-  LOG_PREGEL("f3fd9", DEBUG) << "Storing vertex data (" << numSegments
-                             << " vertices) using " << numT << " threads";
-
-  for (size_t i = 0; i < numT; ++i) {
-    SchedulerFeature::SCHEDULER->queue(RequestLane::INTERNAL_LOW, [=, this] {
-      size_t startI = i * (numSegments / numT);
-      size_t endI = (i + 1) * (numSegments / numT);
-      TRI_ASSERT(endI <= numSegments);
-
-      try {
-        RangeIterator<Vertex<V, E>> it = vertexIterator(startI, endI);
-        storeVertices(_config->globalShardIDs(), it, i, statusUpdateCallback);
-        // TODO can't just write edges with SmartGraphs
-      } catch (std::exception const& e) {
-        LOG_PREGEL("e22c8", ERR) << "Storing vertex data failed: " << e.what();
-      } catch (...) {
-        LOG_PREGEL("51b87", ERR) << "Storing vertex data failed";
-      }
-
-      uint32_t numRunning =
-          _runningThreads.fetch_sub(1, std::memory_order_relaxed);
-      _feature.metrics()->pregelNumberOfThreads->fetch_sub(1);
-      TRI_ASSERT(numRunning > 0);
-      if (numRunning - 1 == 0) {
-        LOG_PREGEL("b5a21", DEBUG)
-            << "Storing data took " << (TRI_microtime() - now) << "s";
-        cb();
-      }
-    });
-  }
+  _runningThreads.store(1);
+  _feature.metrics()->pregelNumberOfThreads->fetch_add(1);
+
+  LOG_PREGEL("f3fd9", DEBUG)
+      << "Storing vertex data (" << _quiver.numberOfVertices()
+      << " vertices) using " << 1 << " threads";
+
+  SchedulerFeature::SCHEDULER->queue(RequestLane::INTERNAL_LOW, [=, this] {
+    try {
+      storeVertices(_config->globalShardIDs(), statusUpdateCallback);
+      // TODO can't just write edges with SmartGraphs
+    } catch (std::exception const& e) {
+      LOG_PREGEL("e22c8", ERR) << "Storing vertex data failed: " << e.what();
+    } catch (...) {
+      LOG_PREGEL("51b87", ERR) << "Storing vertex data failed";
+    }
+    uint32_t numRunning =
```

Comment:
I don't think we need this variable anymore because we have only one thread now, or not?